From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LibreScoot <dev@librescoot.org>
Date: Mon, 28 Apr 2025 11:00:00 +0200
Subject: [PATCH] Custom boot logo for LibreScoot

This patch adds a custom boot logo for LibreScoot devices,
with a resolution of 480x480 pixels.

Signed-off-by: LibreScoot <dev@librescoot.org>
---
 drivers/video/logo/Kconfig       |  7 +++++++
 drivers/video/logo/Makefile      |  1 +
 drivers/video/logo/logo_custom_clut224.ppm | 0
 3 files changed, 8 insertions(+), 0 deletions(-)
 create mode 100644 drivers/video/logo/logo_custom_clut224.ppm

diff --git a/drivers/video/logo/Kconfig b/drivers/video/logo/Kconfig
index xxxxxxx..xxxxxxx 100644
--- a/drivers/video/logo/Kconfig
+++ b/drivers/video/logo/Kconfig
@@ -72,4 +72,11 @@ config LOGO_LINUX_MONO
 	  For a monochrome penguin logo at the top of your screen, say Y
 	  here.
 
+config LOGO_CUSTOM_CLUT224
+	bool "480x480 Custom logo"
+	depends on LOGO
+	default y
+	help
+	  Custom logo for LibreScoot devices, with 480x480 resolution.
+
 endif # LOGO
diff --git a/drivers/video/logo/Makefile b/drivers/video/logo/Makefile
index xxxxxxx..xxxxxxx 100644
--- a/drivers/video/logo/Makefile
+++ b/drivers/video/logo/Makefile
@@ -3,3 +3,4 @@ obj-$(CONFIG_LOGO_LINUX_MONO)		+= logo_linux_mono.o
 obj-$(CONFIG_LOGO_LINUX_VGA16)		+= logo_linux_vga16.o
 obj-$(CONFIG_LOGO_LINUX_CLUT224)	+= logo_linux_clut224.o
 obj-$(CONFIG_LOGO_BLACKFIN_CLUT224)	+= logo_blackfin_clut224.o
+obj-$(CONFIG_LOGO_CUSTOM_CLUT224)	+= logo_custom_clut224.o
diff --git a/drivers/video/logo/logo_custom_clut224.ppm b/drivers/video/logo/logo_custom_clut224.ppm
new file mode 100644
index 0000000..0000000
--- /dev/null
+++ b/drivers/video/logo/logo_custom_clut224.ppm
@@ -0,0 +1,1 @@
# Placeholder - actual logo file will be provided in recipe
-- 
2.25.1
